op-devstack/shim/network.go

```diff
@@ -50,7 +50,9 @@ func (p *presetNetwork) Faucets() []stack.Faucet {
 
 func (p *presetNetwork) Faucet(m stack.FaucetMatcher) stack.Faucet {
 	v, ok := findMatch(m, p.faucets.Get, p.Faucets)
-	p.require().True(ok, "must find faucet %s", m)
+	if !ok {
+		return nil
+	}
```

Comment:
Just want to confirm, do you think this is fine?
cc: @protolambda , @sigma

Reply:
that doesn't seem ideal, it's more comfortable for the tests to be able to assume that this interface is provided to them.

That was the reason behind having op-acceptor transparently provide one if the devnet doesn't expose it already.
So I'm curious, what use-case would necessitate that change?

Reply:
Yeah, I understand that but what led me to it was to eventually make Faucet to be an optional presence in the network.

Rationale for that: a network can be healthy, functional and well-alive without a faucet (as opposed to say a conductor or a batcher) despite it being a nice-to-have in certain situations. Though I am not sure behind the idea/motivation of the initial development of presetNetwork, my first impression of it was it being a representation of a network i.e. it enforcing what's necessary to run a network, and rest optional.

Hope that makes sense. Happy to be corrected and learn something new around this.

Reply:
Thanks!